nix/t1/elaborate.nix

```diff
@@ -28,30 +29,31 @@ let
     "--ip-config"
     # Can't use `toString` here, or due to some shell escape issue, Java nio cannot find the path
     "${elaborate-config}"
-    "--target-dir"
-    "elaborate"
   ] ++ lib.optionals (lib.elem target [ "subsystem" "subsystememu" "fpga" ]) [ "--rvopcodes-path" "${riscv-opcodes-src}" ];
 in
 stdenvNoCC.mkDerivation {
   name = "t1-${config-name}-elaborate-${target}";
   nativeBuildInputs = [ jq espresso circt ];
   buildCommand = ''
     mkdir -p elaborate $out
-    ${elaborator}/bin/elaborator ${target} ${lib.escapeShellArgs elaborateArgs}
+    ${elaborator}/bin/elaborator ${target} ${lib.escapeShellArgs elaborateArgs} "--target-dir" ${if use-binder then "$out" else "elaborate"} ${lib.optionalString (use-binder) "--use-binder"}
 
-    # TODO: use binder to replace emiting below
-    firtool elaborate/*.fir --annotation-file elaborate/*.anno.json -o $out ${mfcArgs}
+    if [ ${lib.boolToString use-binder} = false]; then
```

Comment:
Move the condition from bash to nix. Use `lib.optionalString` to produce empty string when use-binder = false.